jerry-core/jerry.c

```diff
@@ -635,6 +635,24 @@ jerry_api_release_object (jerry_api_object_t *object_p) /**< pointer acquired th
 } /* jerry_api_release_object */
 
 /**
+ * Acquire specified Jerry API value
+ */
+jerry_api_value_t *
+jerry_api_acquire_value (jerry_api_value_t *value_p) /**< API value */
+{
+  jerry_assert_api_available ();
+
+  if (value_p->type == JERRY_API_DATA_TYPE_STRING)
+  {
+    jerry_api_acquire_string (value_p->u.v_string);
+  }
+  else if (value_p->type == JERRY_API_DATA_TYPE_OBJECT)
+  {
+    jerry_api_acquire_object (value_p->u.v_object);
+  }
```

Comment:
The type of a jerry API value (given with the `jerry_api_data_type_t` enum) can be different from string or object. The function does not deal with them (there is no final `else` branch) nor does it document what the expected outcome on such a value is.